src/main/common/draft/draftMiddleware.ts

```diff
@@ -1,41 +1,60 @@
 import * as express from 'express'
-import DraftStoreClient from 'common/draft/draftStoreClient'
 
-export class DraftMiddleware<T> {
-  client: DraftStoreClient<T>
+import { UUIDUtils } from 'common/utils/uuidUtils'
+import { Draft } from '@hmcts/draft-store-client/dist/app/models/draft'
+import { DraftService } from 'services/draftService'
 
-  constructor (public draftType: string, public deserializeFn: (value: any) => T = (value) => value) {
-    if (!draftType || draftType.trim() === '') {
-      throw new Error('Draft type is required to instantiate middleware')
-    }
-    this.client = new DraftStoreClient<T>(draftType)
+/**
+ * Filters list of drafts to return only these matching external ID. If none of the drafts has external ID set
+ * then unchanged list is returned so that they can be migrated to new format with external ID (legacy drafts scenario).
+ */
+function tryFilterByExternalId<T> (drafts: Draft<T>[], externalId: string): Draft<T>[] {
+  if (drafts.filter(draft => draft.document['externalId'] !== undefined).length === 0) {
+    return drafts
   }
 
-  retrieve (res: express.Response, next: express.NextFunction): void {
-    if (res.locals.isLoggedIn) {
-      this.client
-        .retrieve(res.locals.user.id, this.deserializeFn)
-        .then(draft => {
-
-          if (!draft) {
-            draft = this.deserializeFn(null)
-          }
-          res.locals.user[`${this.draftType}Draft`] = draft
-          next()
-        })
-        .catch(next)
-    } else {
-      next()
-    }
-  }
+  return drafts.filter(item => item.document['externalId'] === externalId)
+}
 
-  save (res: express.Response, next: express.NextFunction): Promise<void> {
-    return this.client
-      .save(res.locals.user.id, res.locals.user[`${this.draftType}Draft`])
-  }
+export class DraftMiddleware {
 
-  delete (res: express.Response, next: express.NextFunction): Promise<void> {
-    return this.client
-      .delete(res.locals.user.id)
+  static requestHandler<T> (draftType: string, deserializeFn: (value: any) => T = (value) => value): express.RequestHandler {
+    return async function (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+      if (res.locals.isLoggedIn) {
+        try {
+          new DraftService()['find'](draftType, '100', res.locals.user.bearerToken, deserializeFn)
```

Comment:
this is also wrong 
```
new DraftService().find(...)
```

Reply:
We are calling JavaScript functions here

Reply:
still wrong and shouldn't be doing it